klippa/src/lib.rs

```diff
@@ -549,18 +558,79 @@ impl Plan {
-            //TODO: generate varstore innermaps or something similar
+
+            if variation_indices.is_empty() {
+                return;
+            }
+            // generate 3 maps:
+            // colr_varidx_delta_map
+            // When delta set index map is not included, it's a mapping from varIdx-> (new varIdx,delta).
+            // Otherwise, it's a mapping from old delta set idx-> (new delta set idx, delta).
+            // Mapping delta set indices is the same as gid mapping.
+            //
+            // colr_varstore_inner_maps:
+            // mapping from old varidx -> new varidx
+            //
+            // colr_new_deltaset_idx_varidx_map:
+            // generate new delta set idx-> new var_idx map if DeltsSetIndexMap exists
+            if let Some(Ok(var_store)) = colr.item_variation_store() {
+                let vardata_count = var_store.item_variation_data_count() as u32;
+                let Ok(var_index_map) = colr.var_index_map().transpose() else {
+                    return;
+                };
+
+                let mut delta_set_indices = IntSet::empty();
+                let mut deltaset_idx_var_idx_map = FnvHashMap::default();
+                // when a DeltaSetIndexMap is included, collected variation indices are actually delta set indices,
+                // we need to map them into variation indices
+                if var_index_map.is_some() {
+                    let var_index_map = var_index_map.as_ref().unwrap();
+                    delta_set_indices.extend(variation_indices.iter());
+                    variation_indices.clear();
+                    for idx in delta_set_indices.iter() {
+                        if let Ok(var_idx) = var_index_map.get(idx) {
+                            let var_idx = ((var_idx.outer as u32) << 16) + var_idx.inner as u32;
+                            variation_indices.insert(var_idx);
+                            deltaset_idx_var_idx_map.insert(idx, var_idx);
+                        }
+                    }
+                }
+                remap_variation_indices(
+                    vardata_count,
+                    &variation_indices,
+                    &mut self.colr_varidx_delta_map,
+                );
+                generate_varstore_inner_maps(
+                    &variation_indices,
+                    vardata_count,
+                    &mut self.colr_varstore_inner_maps,
+                );
+
+                // if DeltaSetIndexMap exists, we need to use deltaset index instead of var_idx
+                if var_index_map.is_some() {
+                    let (mut new_deltaset_idx_varidx_map, mut deltaset_idx_delta_map) =
+                        remap_delta_set_indices(
+                            &delta_set_indices,
+                            &deltaset_idx_var_idx_map,
+                            &self.colr_varidx_delta_map,
+                        );
+                    std::mem::swap(
```

Comment:
if we aren't going to use the old value, it would be more idiomatic to use `mem::replace` instead of `mem::swap`.